core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1411,7 +1411,7 @@ class ReplicaManager(val config: KafkaConfig,
             } else {
               warn(s"Leader $localBrokerId failed to record follower $followerId's position " +
                 s"${readResult.info.fetchOffsetMetadata.messageOffset} since the replica is not recognized to be " +
-                s"one of the assigned replicas ${partition.assignedReplicas.map(_.brokerId).mkString(",")} " +
+                s"one of the assigned replicas ${partition.remoteReplicas.map(_.brokerId).mkString(",")} " +
```

Comment:
There may be a few cases like this where we used to be referring to the full replica set. It might be helpful to have a def like the following:
```scala
def allReplicas: Set[Int] = remoteReplicas + leaderId
```

Reply:
Done.